File: frontend/src/models/Seat.ts
import { Flight } from './Flight';

export interface Seat {
    id: number;
    flight?: Flight;
    seatNumber: string;
    row: number;
    column: string;
    isWindow: boolean;
    isAisle: boolean;
    isEmergencyExit: boolean;
    hasExtraLegroom: boolean;
    isOccupied: boolean;
    seatClass: string;
    isRecommended?: boolean;
}

export interface SeatPreferences {
    numberOfSeats: number;
    preferWindow: boolean;
    preferAisle: boolean;
    preferExtraLegroom: boolean;
    seatsTogetherRequired: boolean;
}

File: src/main/java/com/cgi/flightbooking/models/Seat.java
package com.cgi.flightbooking.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Seat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // The flight this seat belongs to
    @ManyToOne
    private Flight flight;

    private String seatNumber; // e.g., "12A"
    private int row;
    private String column; // e.g., "A", "B", etc.

    private boolean isWindow;
    private boolean isAisle;
    private boolean isEmergencyExit;
    private boolean hasExtraLegroom;

    private boolean isOccupied;
    private String seatClass; // "ECONOMY", "BUSINESS", "FIRST"
}
